From ed4959f6b15f7db6e0c52dc1ec2f35e90bed47be Mon Sep 17 00:00:00 2001
Message-Id: <ed4959f6b15f7db6e0c52dc1ec2f35e90bed47be.1646986976.git.thomas@gllm.fr>
In-Reply-To: <2f52ccaa636f6e528fc9962b071236ca3c0753ca.1646986976.git.thomas@gllm.fr>
References: <2f52ccaa636f6e528fc9962b071236ca3c0753ca.1646986976.git.thomas@gllm.fr>
From: Thomas Guillem <thomas@gllm.fr>
Date: Tue, 25 Jan 2022 10:12:51 +0100
Subject: [PATCH 22/23] dsm: connect the session in login()

(cherry picked from commit 7dd7780d8fca179b93ec4eb57e9da3c08333af5f)
Signed-off-by: Thomas Guillem <thomas@gllm.fr>
---
 modules/access/dsm/access.c | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/modules/access/dsm/access.c b/modules/access/dsm/access.c
index 795872092e..220b3ff7dc 100644
--- a/modules/access/dsm/access.c
+++ b/modules/access/dsm/access.c
@@ -171,18 +171,6 @@ static int Open( vlc_object_t *p_this )
     msg_Dbg( p_access, "Session: Host name = %s, ip = %s", p_sys->netbios_name,
              inet_ntoa( p_sys->addr ) );
 
-    /* Now that we have the required data, let's establish a session */
-    status = smb_session_connect( p_sys->p_session, p_sys->netbios_name,
-                                  p_sys->addr.s_addr, SMB_TRANSPORT_TCP );
-    if( status != DSM_SUCCESS )
-    {
-        msg_Err( p_access, "Unable to connect/negotiate SMB session");
-        /* FIXME: libdsm wrongly return network error when the server can't
-         * handle the SMBv1 protocol */
-        status = DSM_ERROR_GENERIC;
-        goto error;
-    }
-
     get_path( p_access );
 
     if( login( p_access ) != VLC_SUCCESS )
@@ -382,6 +370,17 @@ static int login( stream_t *p_access )
     }
     psz_domain = credential.psz_realm ? credential.psz_realm : p_sys->netbios_name;
 
+    /* Now that we have the required data, let's establish a session */
+    int status = smb_session_connect( p_sys->p_session, p_sys->netbios_name,
+                                      p_sys->addr.s_addr, SMB_TRANSPORT_TCP );
+    if( status != DSM_SUCCESS )
+    {
+        msg_Err( p_access, "Unable to connect/negotiate SMB session");
+        /* FIXME: libdsm wrongly return network error when the server can't
+         * handle the SMBv1 protocol */
+        goto error;
+    }
+
     /* Try to authenticate on the remote machine */
     int connect_err = smb_connect( p_access, psz_login, psz_password, psz_domain );
     if( connect_err == ENOENT )
-- 
2.30.2

